addons/website_slides/models/res_partner.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    slide_channels_count = fields.Integer('Followed Course Count', compute='_compute_channels_count')
+    company_slide_channels_count = fields.Integer('Followed Course By Company Partners Count', compute='_compute_company_channels_count')
+
+    def _compute_channels_count(self):
+        read_group_res = self.env['slide.channel.partner'].sudo().read_group(
+            [('partner_id', 'in', self.ids)],
+            ['partner_id'], 'partner_id'
+        )
+        data = dict((res['partner_id'][0], res['partner_id_count']) for res in read_group_res)
+        for partner in self:
+            partner.slide_channels_count = data.get(partner.id, 0)
+
+    def _compute_company_channels_count(self):
+        search_res = self.env['slide.channel'].sudo().search(
+            [('partner_ids', 'in', self.child_ids.ids)]
+        )
+        self.company_slide_channels_count = len(search_res)
+
+    def action_redirect_to_followed_channels(self):
```

Comment:
I would rename to action_view_courses a bit shorted (and action_view even if not in guidelines could become a guideline :p).